### ISSUE TITLE
Support async and await for Clock.SetThreadUtcNow()

### ISSUE THREAD

**Issue by Sjaaky** [Qowaiv/Qowaiv#303]
Currently `Clock.SetTimeForCurrentThread` stores the `Func<DateTime>` in a `[ThreadStatic]` variable. This works perfectly for synchronous tests. 
```cs
using(Clock.SetTimeForCurrentThread(() => new DateTime(2023, 2, 17))
{
    // test code
}
```
But for tests that use async/await, code is executed on different threads. On those threads the Clock is not set.
To solve this, the `threadUtcNow` and `threadTimeZone` should be stored using `AsyncLocal<>`.

Question is, should the name of `SetTimeForCurrentThread` be changed? I can't think of a better name than the current name, it applies to more situations though.

**Comment by Corniel** [comment]
`ForCurrentContext` is a good new suffix. However, that should then obviously only be a new way of doing it, and the old methods should be marked as depreciated.

**Comment by Corniel** [comment]
Solved with #304
